tests/integration/targets/pacman/tasks/package_urls.yml

```
@@ -62,54 +97,89 @@
           - '{{url_pkg}}'
           - '{{file_pkg}}'
       register: install_4
+    - name: Delete cached {{url_pkg}}
+      file:
+        path: '{{url_pkg_filename}}'
+        state: absent
 
     - name: Install new package with already installed packages from mixed sources
       pacman:
         name:
           - '{{reg_pkg}}'
-          - '{{url_pkg_url.stdout}}'
+          - '{{url_pkg_url}}'
           - '{{file_pkg_path}}'
           - '{{extra_pkg}}'
       register: install_5
+    - name: Delete cached {{url_pkg}}
+      file:
+        path: '{{url_pkg_filename}}'
+        state: absent
 
     - name: Uninstall packages - mixed sources (check mode)
       pacman:
         state: absent
         name:
           - '{{reg_pkg}}'
-          - '{{url_pkg_url.stdout}}'
+          - '{{url_pkg_url}}'
           - '{{file_pkg_path}}'
       check_mode: True
       register: uninstall_1
+    - name: Uninstall packages - url (check mode, cached)
+      pacman:
+        state: absent
+        name:
+          - '{{url_pkg_url}}'
+      check_mode: True
+      register: uninstall_1c
+    - name: Delete cached {{url_pkg}}
+      file:
+        path: '{{url_pkg_filename}}'
+        state: absent
 
     - name: Uninstall packages - mixed sources
       pacman:
         state: absent
         name:
           - '{{reg_pkg}}'
-          - '{{url_pkg_url.stdout}}'
+          - '{{url_pkg_url}}'
           - '{{file_pkg_path}}'
       register: uninstall_2
+    - name: Delete cached {{url_pkg}}
+      file:
+        path: '{{url_pkg_filename}}'
+        state: absent
 
     - name: Uninstall packages - mixed sources (idempotency)
       pacman:
         state: absent
         name:
           - '{{reg_pkg}}'
-          - '{{url_pkg_url.stdout}}'
+          - '{{url_pkg_url}}'
           - '{{file_pkg_path}}'
       register: uninstall_3
 
+    - name: Uninstall package - url (idempotency, cached)
+      pacman:
+        state: absent
+        name:
+          - '{{url_pkg_url}}'
+      register: uninstall_3c
+
     - assert:
         that:
           - install_1 is changed
           - install_1.msg == 'Would have installed 3 packages'
           - install_1.packages|sort() == [reg_pkg, url_pkg, file_pkg]|sort()
+          - install_1c is changed
+          - install_1c.msg == 'Would have installed 3 packages'
+          - install_1c.packages|sort() == [reg_pkg, url_pkg, file_pkg]|sort()
```

There's only one package installed in that step
```suggestion
          - install_1c.msg == 'Would have installed 1 packages'
          - install_1c.packages|sort() == [url_pkg]
```